osu.Game/Database/LegacyModelExporter.cs

```diff
@@ -0,0 +1,164 @@
+                    .GetFiles(string.Empty, $"{itemFilename}*{FileExtension}")
+                    .Concat(exportStorage.GetDirectories(string.Empty));
+            string filename = NamingUtils.GetNextBestFilename(existingExports, $"{itemFilename}{FileExtension}");
+            bool success = false;
+
+            ProgressNotification notification = new ProgressNotification
+            {
+                State = ProgressNotificationState.Active,
+                Text = "Exporting...",
+                CompletionText = "Export completed"
+            };
+            notification.CompletionClickAction += () => exportStorage.PresentFileExternally(filename);
+            PostNotification?.Invoke(notification);
+
+            try
+            {
+                using (var stream = exportStorage.CreateFileSafely(filename))
+                {
+                    success = await ExportToStreamAsync(model, stream, notification, cancellationToken ?? notification.CancellationToken).ConfigureAwait(false);
+                }
+            }
+            catch (OperationCanceledException)
+            {
+                success = false;
+            }
+            finally
+            {
+                // cleanup if export is failed or canceled.
+                if (!success)
+                {
+                    exportStorage.Delete(filename);
+                }
+
+                exportingModels.Remove(model);
+            }
+        }
+
+        /// <summary>
+        /// Export model to stream.
+        /// </summary>
+        /// <param name="model">The medel which have <see cref="IHasGuidPrimaryKey"/>.</param>
+        /// <param name="stream">The stream to export.</param>
+        /// <param name="notification">The notification will displayed to the user</param>
+        /// <param name="cancellationToken">The Cancellation token that can cancel the exporting.</param>
+        /// <returns>Whether the export was successful</returns>
+        public async Task<bool> ExportToStreamAsync(TModel model, Stream stream, ProgressNotification? notification = null, CancellationToken cancellationToken = default)
+        {
+            ProgressNotification notify = notification ?? new ProgressNotification();
+
+            Guid id = model.ID;
+            return await Task.Run(() =>
+            {
+                realmAccess.Run(r =>
+                {
+                    TModel refetchModel = r.Find<TModel>(id);
+                    ExportToStream(refetchModel, stream, notify, cancellationToken);
+                });
+            }, cancellationToken).ContinueWith(t =>
+            {
+                if (t.IsFaulted)
+                {
+                    notify.State = ProgressNotificationState.Cancelled;
+                    Logger.Error(t.Exception, "An error occurred while exporting");
+                    return false;
+                }
+
+                notify.CompletionText = "Export Complete, Click to open the folder";
```

Comment:
Incorrect grammar. Also it doesn't really open the folder but present the file.

Maybe look at https://github.com/ppy/osu/blob/f8537c1cbe98bc38f0423e8eabdc2332a21d22ab/osu.Game/Database/RealmArchiveModelImporter.cs#L157-L172 for inspiration.